drivers/net/nsos_sockets.c

```diff
@@ -1608,9 +1624,26 @@ static void nsos_delayed_connect_fn(struct k_work *work)
 {
 	struct k_work_delayable *dwork = k_work_delayable_from_work(work);
 	struct nsos_conn_data *data = CONTAINER_OF(dwork, struct nsos_conn_data, work);
+	struct conn_mgr_conn_binding *const binding = conn_mgr_if_get_binding(data->iface);
 
 	LOG_INF("NSOS: active");
 	net_if_dormant_off(data->iface);
+
+	/* Start the idle timeout */
+	if (binding->idle_timeout > CONN_MGR_IF_NO_TIMEOUT) {
+		k_work_reschedule(&data->idle_work, K_SECONDS(binding->idle_timeout));
+	}
+}
+
+static void nsos_idle_fn(struct k_work *work)
```

Comment:
Hmm, so from the issue description and the new documentation I had an impression that this idle timeout handling would be a generic implementation in the conn_mgr, and not handled individually by each binding implementation?

So that when the timer for particular binding expires, the conn_mgr would call `conn_mgr_if_disconnect()` on the interface. This way the feature would practically become avaialble for each binding implementation for free? That would also put the need of having `used()` API in `struct conn_mgr_conn_api` at question.

Or do you forsee there may be some differences between implementations and it is actually beneficial to have the incactivity timer to be implemented by each binding individually?

Reply:
done